nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,342 @@
+    return accuracy_aware_loop.restore_accuracy(model, initial_metric,
+                                                quantized_model, quantized_metric,
+                                                validation_dataset, validation_fn, max_drop)
+
+
+def _create_message(nodes: Iterable[NNCFNode]) -> str:
+    names = [f'\t{x.node_name}' for x in nodes]
+    return '\n'.join(names)
+
+
+class AccuracyAwareLoop:
+
+    def __init__(self,
+                 ranking_subset_size: int = 300,
+                 max_num_iterations: int = sys.maxsize,
+                 is_native: bool = True):
+        """
+        :param ranking_subset_size:
+        :param max_num_iterations:
+        :param is_native:
+        """
+        self.ranking_subset_size = ranking_subset_size
+        self.max_num_iterations = max_num_iterations
+        # TODO(andrey-churkin): Should be removed when native implementation
+        # will become the main one.
+        self.is_native = is_native
+
+    def restore_accuracy(self,
+                         initial_model: TModel,
+                         initial_metric: float,
+                         quantized_model: TModel,
+                         quantized_metric: float,
+                         validation_dataset: Dataset,
+                         validation_fn: Callable[[Any, Iterable[Any]], float],
+                         max_drop: float = 0.01) -> TModel:
+        """
+        Restores the accuracy of the quantized model by removing the groups of quantizers
+        that contribute the most to the drop in accuracy.
+
+        :param initial_model: Initial model (not quantized).
+        :param initial_metric: Metric value for initial model.
+        :param quantized_model: Quantized model.
+        :param quantized_metric: Metric value for quantized model.
+        :param validation_dataset: A dataset for the validation process.
+        :param validation_fn: A validation function to validate the model. It should take
+            two argumets:
+            - `model`: model to be validate.
+            - `validation_dataset`: dataset that provides data items to
+                validate the provided model.
+            The function should return the value of the metric with the following meaning:
+            A higher value corresponds to better performance of the model.
+        :param max_drop: The maximum absolute accuracy drop that should be achieved.
+        :return: The quantized model whose metric `final_metric` is satisfied the following condition
+
+            initial_metric - final_metric <= max_drop.
+        """
+        accuracy_drop = initial_metric - quantized_metric
+        nncf_logger.info(f'Accuracy drop: {accuracy_drop}')
+
+        if accuracy_drop <= max_drop:
+            return quantized_model
+
+        nncf_logger.info('Changing the scope of quantizer nodes.')
+
+        # Hyperparameters of algorithm
+        USE_PREVIOUS_IF_DROP_INCREASE = True
```

Comment:
Why do we need this variable since it's used only once and can't be changed outside the pipeline?

Reply:
Done